rhel/rhcc/updater.go

```diff
@@ -0,0 +1,248 @@
+package rhcc
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"io"
+	"net/http"
+	"sort"
+
+	"github.com/quay/zlog"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/libvuln/driver"
+	"github.com/quay/claircore/pkg/cpe"
+	"github.com/quay/claircore/pkg/rhctag"
+	"github.com/quay/claircore/pkg/tmp"
+	"github.com/quay/claircore/rhel"
+)
+
+const (
+	dbURL = "https://access.redhat.com/security/data/metrics/cvemap.xml"
+)
+
+var (
+	_ driver.Updater      = (*updater)(nil)
+	_ driver.Configurable = (*updater)(nil)
+)
+
+// updater fetches and parses cvemap.xml
+type updater struct {
+	client *http.Client
+	url    string
+}
+
+type UpdaterConfig struct {
+	URL string `json:"url" yaml:"url"`
+}
+
+const updaterName = "rhel-container-updater"
+
+func (*updater) Name() string {
+	return updaterName
+}
+
+func UpdaterSet(_ context.Context) (driver.UpdaterSet, error) {
+	us := driver.NewUpdaterSet()
+	if err := us.Add(&updater{}); err != nil {
+		return us, err
+	}
+	return us, nil
+}
+
+func (u *updater) Configure(ctx context.Context, f driver.ConfigUnmarshaler, c *http.Client) error {
+	u.url = dbURL
+	u.client = c
+	var cfg UpdaterConfig
+	if err := f(&cfg); err != nil {
+		return err
+	}
+	if cfg.URL != "" {
+		u.url = cfg.URL
+	}
+	return nil
+}
+
+func (u *updater) Fetch(ctx context.Context, hint driver.Fingerprint) (io.ReadCloser, driver.Fingerprint, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "rhel/rhcc/Updater.Fetch")
+
+	zlog.Info(ctx).Str("database", u.url).Msg("starting fetch")
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, u.url, nil)
+	if err != nil {
+		return nil, hint, fmt.Errorf("rhcc: unable to construct request: %w", err)
+	}
+
+	if hint != "" {
+		zlog.Debug(ctx).
+			Str("hint", string(hint)).
+			Msg("using hint")
+		req.Header.Set("if-none-match", string(hint))
+	}
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, hint, fmt.Errorf("rhcc: error making request: %w", err)
+	}
+	defer res.Body.Close()
+
+	switch res.StatusCode {
+	case http.StatusOK:
+		// break
```

Comment:
We changed these all recently to still check on 200, should we also change this or are we confident that https://access.redhat.com/security/data/metrics/cvemap.xml is going to return correctly?